PR: https://github.com/odoo/odoo/pull/233963

From: 51489bd94e3fed37df5609f1cce6c878d2e33b50
From: Aurélien Warnon
Date: 2026-02-14 01:01:28

Breaking data model changes scores: del:7 + add:0, change matches:
-    _inherit = 'crm.lead'
-    origin_survey_id = fields.Many2one('survey.survey', string='Survey', index='btree_not_null', ondelete='set null')
-    _inherit = 'crm.team'
-    origin_survey_ids = fields.One2many(
-    lead_ids = fields.One2many('crm.lead', 'origin_survey_id')

Total Changes: 44

[IMP] survey_crm: cleanup survey_crm "origin" fields

RATIONALE

The way we currently use the UTM trackers (aka source/medium/campaign) does not
comply with the industry standards, leading to confusion and globally unclear
or not usable reporting on UTM data.

PURPOSE

Now that we have a global concept of UTM reference, we can use it to replace
the recently introduced field: origin_survey_id.

By adapting the various domain and actions, we end up getting the same
functional result without requiring this field.

(Note: "origin_survey_ids" was unused and cleaned at the same time).

Task-3330036

Part-of: odoo/odoo#233963
Related: odoo/enterprise#98569
Related: odoo/upgrade#8746
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/survey_crm/models/__init__.py
+++ b/addons/survey_crm/models/__init__.py
@@ -1,5 +1,3 @@
-from . import crm_lead
-from . import crm_team
 from . import survey_question
 from . import survey_question_answer
 from . import survey_survey

--- a/addons/survey_crm/models/crm_lead.py
+++ b/None
@@ -1,9 +0,0 @@
-from odoo import fields, models
-
-
-class CrmLead(models.Model):
-    _inherit = 'crm.lead'
-
-    # Origin of the lead
-    # TODO awa: check if still necessary, could be replace with utm.reference
-    origin_survey_id = fields.Many2one('survey.survey', string='Survey', index='btree_not_null', ondelete='set null')

--- a/addons/survey_crm/models/crm_team.py
+++ b/None
@@ -1,10 +0,0 @@
-from odoo import fields, models
-
-
-class CrmTeam(models.Model):
-    _inherit = 'crm.team'
-
-    # Origin of leads
-    origin_survey_ids = fields.One2many(
-        'survey.survey', 'team_id',
-        string='Survey opportunities related to the sales team')

--- a/addons/survey_crm/models/survey_survey.py
+++ b/addons/survey_crm/models/survey_survey.py
@@ -8,7 +8,6 @@ class SurveySurvey(models.Model):
 
     generate_lead = fields.Boolean('Lead Generating', compute='_compute_generate_lead', store="True")
     lead_count = fields.Integer('Leads', help='Number of leads created by this survey', compute='_compute_lead_count')
-    lead_ids = fields.One2many('crm.lead', 'origin_survey_id')  # Leads created by the survey
     team_id = fields.Many2one(
         'crm.team', string='Assign Leads to',
         index='btree_not_null', ondelete='set null')  # If a submission generates an opportunity, this sales team will be associated with it
@@ -19,17 +18,14 @@ class SurveySurvey(models.Model):
             survey.generate_lead = survey.survey_type in ['survey', 'live_session', 'custom'] and \
                                    any(question_id.generate_lead for question_id in survey.question_ids)
 
-    @api.depends('lead_ids')
     def _compute_lead_count(self):
+        lead_data = self.env['crm.lead'].sudo()._read_group(
+            [('utm_reference', 'in', [f'{survey._name},{survey.id}' for survey in self])],
+            ['utm_reference'], ['__count'],
+        )
+        mapped_data = dict(lead_data)
         for survey in self:
-            if self.ids and self.env['crm.lead'].has_access('read'):
-                leads = self.env['crm.lead']._read_group(
-                    [('origin_survey_id', 'in', self.ids)], ['origin_survey_id'], ['__count'])
-                leads_count_by_survey = {survey.id: count for survey, count in leads}
-                for survey in self:
-                    survey.lead_count = leads_count_by_survey.get(survey.id, 0)
-            else:
-                self.lead_count = 0
+            survey.lead_count = mapped_data.get(f'{survey._name},{survey.id}', 0)
 
     def action_end_session(self):
         ''' Checks if leads need to be created for live sessions (either custom or live_session) '''
@@ -47,5 +43,9 @@ class SurveySurvey(models.Model):
             ast.literal_eval(action.get('context', '{}').strip()),  # ".strip()" prevents a crash of literal_eval which doesn't interpret the "\n" after the dictionary is closed in the string
             create=False,
         )
-        action['domain'] = [('origin_survey_id', 'in', self.ids)]
+        action['domain'] = [(
+            'utm_reference',
+            'in',
+            [f'{survey._name},{survey.id}' for survey in self]
+        )]
         return action

--- a/addons/survey_crm/models/survey_user_input.py
+++ b/addons/survey_crm/models/survey_user_input.py
@@ -60,7 +60,6 @@ class SurveyUser_Input(models.Model):
         return {
             'source_id': self.env['utm.mixin']._utm_ref('utm.utm_source_survey').id,
             'utm_reference': f'{survey._name},{survey.id}',
-            'origin_survey_id': survey.id,
             'team_id': sales_team.id,
             'type': 'opportunity',  # we assume that the lead is sufficiently qualified based on survey responses to be an opportunity
             'user_id': salesperson.id,
